exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/h2/H2Tests.kt

```diff
@@ -9,15 +9,36 @@ import org.jetbrains.exposed.sql.tests.inProperCase
 import org.jetbrains.exposed.sql.tests.shared.assertEquals
 import org.jetbrains.exposed.sql.transactions.TransactionManager
 import org.jetbrains.exposed.sql.transactions.transactionManager
+import org.jetbrains.exposed.sql.vendors.H2Dialect
+import org.jetbrains.exposed.sql.vendors.currentDialect
 import org.junit.Test
 import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit
 
 class H2Tests : DatabaseTestsBase() {
+    @Test
+    fun testH2VersionIsCorrect() {
+        val systemTestName = System.getProperty("exposed.test.name")
+        withDb(TestDB.ALL_H2) { db ->
+            val dialect = currentDialect
+            if (dialect is H2Dialect) {
```

Comment:
Probably don't need to test every single TestDB.H2, just H2_V2 and H2_V2.

Also, is the check needed? How would it be possible for `dialect` to not be H2Dialect in this case?